### ISSUE TITLE
run/t6246.scala is broken on master

### ISSUE THREAD

**Issue by sjrd** [scala-js/scala-js#2021]
Nightly has been failing consistently over [`run/t6246.scala`](https://github.com/scala/scala/blob/2.12.x/test/files/run/t6246.scala), both in 2.11 and 2.12.

**Comment by sjrd** [comment]
It's a linking error:
```
$ cat t6246-run.log
Referring to non-existent method jl_Class.getInterfaces__Ajl_Class
  called from s_reflect_ClassManifestDeprecatedApis$class.loop$1__p0__s_reflect_ClassTag__sci_Set__sci_Set__jl_Class__Z
  called from s_reflect_ClassManifestDeprecatedApis$class.subtype__p0__s_reflect_ClassTag__jl_Class__jl_Class__Z
  called from s_reflect_ClassManifestDeprecatedApis$class.$$less$colon$less__s_reflect_ClassTag__s_reflect_ClassTag__Z
  called from s_reflect_ClassTag$ClassClassTag.$$less$colon$less__s_reflect_ClassTag__Z
  called from LTest$.testValueClass__s_reflect_ClassTag__V
  called from LTest$.delayedEndpoint$Test$1__V
  called from LTest$delayedInit$body.apply__O
  called from s_Function0$class.apply$mcV$sp__F0__V
  called from sr_AbstractFunction0.apply$mcV$sp__V
  called from s_App$class.$$anonfun$1__p0__s_App__F0__V
  called from s_App$class.main__s_App__AT__V
  called from LTest$.main__AT__V
  called from LPartestLauncher$.launch
  exported to JavaScript with @JSExport
involving instantiated classes:
  s_reflect_ClassTag$ClassClassTag
  LTest$
    (already seen, not repeating call stack)
  LTest$delayedInit$body
    (already seen, not repeating call stack)
  sjsr_AnonFunction0
    (already seen, not repeating call stack)
  LPartestLauncher$
    (already seen, not repeating call stack)
Exception in thread "main" java.lang.RuntimeException: There were linking errors
        at scala.sys.package$.error(package.scala:27)
        at org.scalajs.core.tools.optimizer.Linker.link(Linker.scala:102)
        at org.scalajs.core.tools.optimizer.Linker.link(Linker.scala:72)
        at org.scalajs.core.tools.optimizer.ScalaJSOptimizer$$anonfun$1.apply(ScalaJSOptimizer.scala:119)
        at org.scalajs.core.tools.optimizer.ScalaJSOptimizer$$anonfun$1.apply(ScalaJSOptimizer.scala:119)
        at org.scalajs.core.tools.optimizer.package$.logTime(package.scala:19)
        at org.scalajs.core.tools.optimizer.ScalaJSOptimizer.optimizeIR(ScalaJSOptimizer.scala:118)
        at org.scalajs.core.tools.optimizer.ScalaJSOptimizer.optimizeIR(ScalaJSOptimizer.scala:99)
        at org.scalajs.core.tools.optimizer.ScalaJSOptimizer$$anonfun$optimizeCP$1.apply$mcV$sp(ScalaJSOptimizer.scala:76)
        at org.scalajs.core.tools.io.CacheUtils$.cached(CacheUtils.scala:41)
        at org.scalajs.core.tools.optimizer.ScalaJSOptimizer.optimizeCP(ScalaJSOptimizer.scala:74)
        at scala.tools.nsc.MainGenericRunner.fastOptimize(MainGenericRunner.scala:93)
        at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:77)
        at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:200)
        at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
```

**Comment by gzm0** [comment]
@sjrd could you please ping Fabien that he fixes the Jenkins emails? These test that fail and we don't notice really shouldn't be that way...
The update they did when I was still at LAMP somehow broke the emails.

**Comment by sjrd** [comment]
This failure stopped last night. A `git bisect` showed that the first "bad" commit (the commit that actually *fixed* the issue) was 7b79765633e54608872ae1aa4772fde8a8e5c2b7
```
7b79765633e54608872ae1aa4772fde8a8e5c2b7 is the first bad commit
commit 7b79765633e54608872ae1aa4772fde8a8e5c2b7
Author: Sébastien Doeraene <sjrdoeraene@gmail.com>
Date:   Sun Oct 11 16:21:58 2015 +0200

    Remove the dependency on j.u.Formatter from Range.description.

    This cuts loose j.u.Formatter entirely from the Reversi example.

:040000 040000 fcad56c8ded8ad9056d060dcb47a87eb877d5e27 209f14c806d494d5545238e91e15dc2a402e9207 M     ci
:040000 040000 c25ee2ce9cbddb2d849dfc248b2b2634aa2d0043 d1c179f3e2c3afeba733eb816050e3f68d641e0c M     scalalib
```
Not sure whether to close this or leave it open for further investigation. Because obviously I have no idea *why* it started to fail nor *why* that commit fixed the issue.

In any case we should close #2029

**Comment by sjrd** [comment]
After careful consideration, I we should blacklist this test anyway. The fact that it *passes* is an accident. `<:<` is not supposed to be supported on `ClassTag`s, as it uses `getInterfaces()`. It happens to pass, I guess, because the set of things reachable from this tiny codebase allows the anlyzer to dce away the portion that needs `getInterfaces()`. But depending on the rest of the codebase, this might not link.
